common/vlog.h

```diff
@@ -13,9 +13,10 @@ namespace Carbon {
 //
 // For example:
 //   CARBON_VLOG() << "Verbose message";
-#define CARBON_VLOG()                 \
-  (vlog_stream_ == nullptr) ? (void)0 \
-                            : CARBON_VLOG_INTERNAL_STREAM(vlog_stream_)
+#define CARBON_VLOG()                             \
+  __builtin_expect(vlog_stream_ == nullptr, true) \
```

Comment:
Should CARBON_CHECK have a similar `__builtin_expect((__VA_ARGS__), false)` ?

Reply:
The `noreturn` provides some help there, but yeah, we can probably do more. Will send another PR.